diff --git a/cmake/libs/RP_FindQt5andKF5.cmake b/cmake/libs/RP_FindQt5andKF5.cmake
index 833b93ef2..c5dfc11ab 100644
--- a/cmake/libs/RP_FindQt5andKF5.cmake
+++ b/cmake/libs/RP_FindQt5andKF5.cmake
@@ -58,15 +58,7 @@ MACRO(FIND_QT5_AND_KF5)
 
 			# Get the plugin directory and Qt prefix.
 			# Prefix will be removed from the plugin directory if necessary.
-			EXECUTE_PROCESS(COMMAND ${QTPATHS5} --plugin-dir OUTPUT_VARIABLE KF5_PLUGIN_INSTALL_DIR)
-			STRING(STRIP "${KF5_PLUGIN_INSTALL_DIR}" KF5_PLUGIN_INSTALL_DIR)
-			IF(NOT KF5_PLUGIN_INSTALL_DIR)
-				MESSAGE(FATAL_ERROR "`qtpaths5` isn't working correctly.")
-			ENDIF(NOT KF5_PLUGIN_INSTALL_DIR)
-			# FIXME: Mageia has the Qt path set to "/usr/lib64/qt5" instead of "/usr".
-			# Reference: https://github.com/GerbilSoft/rom-properties/issues/69
-			INCLUDE(ReplaceHardcodedPrefix)
-			REPLACE_HARDCODED_PREFIX(KF5_PLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}")
+			SET(KF5_PLUGIN_INSTALL_DIR "${KDE_INSTALL_PLUGINDIR}")
 			SET(QT_PLUGIN_INSTALL_DIR "${KF5_PLUGIN_INSTALL_DIR}")
 
 			# Find KF5. (TODO: Version?)
